packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/common/Helpers.cs

```diff
@@ -43,8 +46,35 @@ public static async Task<Compilation> GetCompilationFromDirectoryAsync(
         {
             var directory = GetAssetFileOrDirectoryPath(false, parameters, method, filePath);
             var codeGenAttributeFiles = Path.Combine(_assemblyLocation, "..", "..", "..", "..", "..", "Microsoft.Generator.CSharp.Customization", "src");
-            var workspace = GeneratedCodeWorkspace.CreateExistingCodeProject([directory, codeGenAttributeFiles], Path.Combine(directory, "Generated"));
-            return await workspace.GetCompilationAsync();
+            var project = CreateExistingCodeProject([directory, codeGenAttributeFiles], Path.Combine(directory, "Generated"));
+            var compilation = await project.GetCompilationAsync();
+            return compilation!;
```

Comment:
nit: return it instead of declaring a variable

Reply:
I would like to do that but the return of this method is `Task<Compilation?>` and here we want a `Task<Compilation>`, we must have a local variable and assert it is not null.

Reply:
otherwise it might look like this:
```
return (await project.GetCompilationAsync())!;
```
I prefer local variable rather than the above

Reply:
So it will be null if [SupportsCompilation is false](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.project.getcompilationasync?view=roslyn-dotnet-4.7.0). It's not clear to me when that would happen but perhaps we should add some assertion that it is not null rather than using the null-forgiving operator.

Reply:
Ah I see this is specifically for the test helper, so probably less of an issue here. But for the GetCompilationAsync method we should add validation with a helpful error message that says SupportsCompilation must be true.

Reply:
do you mean that we should add an assertion here on compilation cannot be null?
This method `GetCompilationAsync` is not our method, this method comes from roslyn API.
In our version of `GetCompilationAsync`, we have `Debug.Assert(compilation is not null)` and the method does not return a nullable result.

Reply:
Fixed!